specs/_features/eip7594/p2p-interface.md

```diff
@@ -0,0 +1,196 @@
+# EIP-7594 -- Networking
+
+**Notice**: This document is a work-in-progress for researchers and implementers.
+
+## Table of contents
+
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Modifications in EIP-7594](#modifications-in-eip-7594)
+  - [Preset](#preset)
+  - [Configuration](#configuration)
+  - [Containers](#containers)
+    - [`DataColumnIdentifier`](#datacolumnidentifier)
+  - [Helpers](#helpers)
+      - [`verify_data_column_sidecar_kzg_proofs`](#verify_data_column_sidecar_kzg_proofs)
+      - [`verify_data_column_sidecar_inclusion_proof`](#verify_data_column_sidecar_inclusion_proof)
+      - [`compute_subnet_for_data_column_sidecar`](#compute_subnet_for_data_column_sidecar)
+  - [The gossip domain: gossipsub](#the-gossip-domain-gossipsub)
+    - [Topics and messages](#topics-and-messages)
+      - [Blob subnets](#blob-subnets)
+        - [Deprecated `blob_sidecar_{subnet_id}`](#deprecated-blob_sidecar_subnet_id)
+        - [`data_column_sidecar_{subnet_id}`](#data_column_sidecar_subnet_id)
+  - [The Req/Resp domain](#the-reqresp-domain)
+    - [Messages](#messages)
+      - [DataColumnSidecarsByRoot v1](#datacolumnsidecarsbyroot-v1)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Modifications in EIP-7594
+
+### Preset
+
+| Name | Value | Description |
+| - | - | - |
+| `KZG_COMMITMENTS_INCLUSION_PROOF_DEPTH` | `uint64(floorlog2(get_generalized_index(BeaconBlockBody, 'blob_kzg_commitments')))` (= 4) | <!-- predefined --> Merkle proof index for `blob_kzg_commitments` |
+
+
+### Configuration
+
+*[New in Deneb:EIP4844]*
+
+| Name                                     | Value                             | Description                                                         |
+|------------------------------------------|-----------------------------------|---------------------------------------------------------------------|
+| `MAX_REQUEST_DATA_COLUMN_SIDECARS`       | `NUMBER_OF_COLUMNS` | Maximum number of data column sidecars in a single request  |
+| `MIN_EPOCHS_FOR_DATA_COLUMN_SIDECARS_REQUESTS`  | `2**12` (= 4096 epochs, ~18 days) | The minimum epoch range over which a node must serve data column sidecars  |
```

Comment:
For Deneb, we have `MAX_REQUEST_BLOB_SIDECARS` (768) calculated from `MAX_REQUEST_BLOCKS_DENEB * MAX_BLOBS_PER_BLOCK`

Should `MAX_REQUEST_DATA_COLUMN_SIDECARS` be derived in a similar way, i.e. maximum of 128 blocks of sidecars in a single request: `MAX_REQUEST_BLOCKS_DENEB * NUMBER_OF_COLUMNS` (16384)?

These should probably be added to the config files `configs/mainnet.yaml` and `configs/minimal.yaml` too.